aiokafka/protocol/types.py

```diff
@@ -1,10 +1,31 @@
 import struct
+from collections.abc import Sequence
+from io import BytesIO
 from struct import error
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+    overload,
+)
+
+from typing_extensions import Buffer, TypeAlias
 
 from .abstract import AbstractType
 
+T = TypeVar("T")
 
-def _pack(f, value):
+ValueT: TypeAlias = Union[Type[AbstractType[Any]], "String", "Array", "Schema"]
```

Comment:
Interesting trick. But it would be better to explicitly declare their common interface.

Reply:
Don't mind, I think this solution is good enough for now to go farther with typing. But the target (somewhen in the future) it would be nice to get typing for all parsed messages, e.g. with pydantic-style schemes.